exonum/src/blockchain/mod.rs

```diff
@@ -536,23 +545,23 @@ impl Blockchain {
     }
 
     /// Returns `Connect` messages from peers saved in the cache, if any.
-    pub fn get_saved_peers(&self) -> HashMap<PublicKey, Connect> {
+    pub fn get_saved_peers(&self) -> HashMap<PublicKey, Message<Connect>> {
         let schema = Schema::new(self.snapshot());
         let peers_cache = schema.peers_cache();
         let it = peers_cache.iter().map(|(k, v)| (k, v.clone()));
         it.collect()
     }
 
     /// Saves the given raw message to the consensus messages cache.
-    pub(crate) fn save_message(&mut self, round: Round, raw: &RawMessage) {
-        self.save_messages(round, iter::once(raw.clone()));
+    pub(crate) fn save_message<T: ProtocolMessage>(&mut self, round: Round, raw: Message<T>) {
+        self.save_messages(round, iter::once(raw.into()));
     }
 
-    /// Saves a collection of RawMessage to the consensus messages cache with single access to the
+    /// Saves a collection of SignedMessage to the consensus messages cache with single access to the
     /// `Fork` instance.
     pub(crate) fn save_messages<I>(&mut self, round: Round, iter: I)
     where
-        I: IntoIterator<Item = RawMessage>,
+        I: IntoIterator<Item = Protocol>,
```

Comment:
How you suggest to rename it?

My motivation was:
It's an enum where each variant is a message of our internall `protocol`. Because this enums sums all type of protocol messages - it forms full protocol specification.